homework 4

danildemchenko opened the pull request.

danildemchenko committed:
homework 4

VAmpersand commented on Module_2 Groupe_1/Danil/TabController/ViewController.swift:
```diff
@@ -0,0 +1,162 @@
+    private let greenRectangle = UIView(backgroundColor: .green)
+    
+    //top
+    private let topRedBox = UIView(backgroundColor: .red)
```
Классный вариант локоничного создания view сразу с backgroundColor. Почему же след view объявил по другому? У нас кроме цвета больше ничего не устанавливается для view, значит твой вариант даже лучше. Мог бы все добавить через свое расширение, ничего страшного быть не должно

danildemchenko replied:
а во вслед view не нужно указывать translatesAutoresizingMaskIntoConstraints (я с ними работал через snapKit), поэтому подумал, что излишне будет немного)
Но вообще да, согласен, думаю можно было бы и все остальные view через расширение создавать

VAmpersand commented on Module_2 Groupe_1/Danil/TabController/ViewController.swift:
```diff
@@ -0,0 +1,162 @@
+            [
+                // 16 можно вынести в константу
+                //желтый вью на белый вью
+                yellowRectangle.topAnchor.constraint(equalTo: view.topAnchor, constant: 16),
```
Это верный комментарий, любые константы лучше всего должны быть в константах с внятным названием. Почему же не реализовал?

VAmpersand commented on Module_2 Groupe_1/Danil/TabController/ViewController.swift:
```diff
@@ -0,0 +1,162 @@
+                yellowRectangle.bottomAnchor.constraint(equalTo: view.bottomAnchor, constant: -16),
+                yellowRectangle.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: 16),
+                yellowRectangle.trailingAnchor.constraint(equalTo: view.trailingAnchor, constant: -16),
+                
```
Много отступов, одного достаточно

VAmpersand commented on Module_2 Groupe_1/Danil/TabController/ViewController.swift:
```diff
@@ -0,0 +1,162 @@
+                greenRectangle.centerXAnchor.constraint(equalTo: yellowRectangle.centerXAnchor),
+                greenRectangle.centerYAnchor.constraint(equalTo: yellowRectangle.centerYAnchor),
+                //0.38 можно вынести в константу)
+                greenRectangle.heightAnchor.constraint(equalTo: yellowRectangle.heightAnchor, multiplier: 0.38),
```
Классный вариант констрейнта.

VAmpersand commented on Module_2 Groupe_1/Danil/TabController/ViewController.swift:
```diff
@@ -0,0 +1,162 @@
+                greenRectangle.heightAnchor.constraint(equalTo: yellowRectangle.heightAnchor, multiplier: 0.38),
+                greenRectangle.widthAnchor.constraint(equalTo: greenRectangle.heightAnchor, multiplier: 0.5),
+                
+                
```
Отступ лишний

VAmpersand commented on Module_2 Groupe_1/Danil/TabController/ViewController.swift:
```diff
@@ -0,0 +1,162 @@
+                
+                //красный верхний вью на желтый вью
+                topRedBox.topAnchor.constraint(equalTo: yellowRectangle.topAnchor, constant: 16),
+                topRedBox.centerXAnchor.constraint(equalTo: yellowRectangle.centerXAnchor),
```
Из за этого у тебя на SE отступы больше.

VAmpersand commented on Module_2 Groupe_1/Danil/TabController/ViewController.swift:
```diff
@@ -0,0 +1,162 @@
+                topRedBox.heightAnchor.constraint(equalTo: yellowRectangle.heightAnchor, multiplier: 0.4),
+                topRedBox.widthAnchor.constraint(equalTo: topRedBox.heightAnchor),
+                
+                
```
Отступ

VAmpersand commented on Module_2 Groupe_1/Danil/TabController/ViewController.swift:
```diff
@@ -0,0 +1,162 @@
+        //верхний квадрат
+        topBlueBox.snp.makeConstraints {
+            $0.width.height.equalToSuperview().multipliedBy(0.32)
+            
```
Отступ

VAmpersand commented on Module_2 Groupe_1/Danil/TabController/ViewController.swift:
```diff
@@ -0,0 +1,162 @@
+        
+        //верхний квадрат
+        topBlueBox.snp.makeConstraints {
+            $0.width.height.equalToSuperview().multipliedBy(0.32)
```
$0.size.equalToSuperview().multipliedBy(0.32)

VAmpersand commented on Module_2 Groupe_1/Danil/TabController/ViewController.swift:
```diff
@@ -0,0 +1,162 @@
+        topOrangeRectangle.snp.makeConstraints {
+            $0.width.equalTo(topBlueBox.snp.width).multipliedBy(2)
+            $0.height.equalTo(topBlueBox.snp.width)
+            
```
Отступ

VAmpersand commented on Module_2 Groupe_1/Danil/TabController/ViewController.swift:
```diff
@@ -0,0 +1,162 @@
+            $0.top.leading.equalToSuperview().inset(16)
+        }
+        topOrangeRectangle.snp.makeConstraints {
+            $0.width.equalTo(topBlueBox.snp.width).multipliedBy(2)
```
$0.width.equalTo(topBlueBox).multipliedBy(2) - когда ты приравниваешь параметр к тому же параметру другого вью, то не нужно акцентировать внимание на эталонном параметре. Т.Е. ты устанавливаешь ширину равную ширине topBlueBox, значит не надо дописывать  .snp.width. А вот есть сравниваешь с высотой, то тогда нужно это писать, как ты сделал на след строке. То же самое с другими типами констрейнтов

VAmpersand commented on Module_2 Groupe_1/Danil/TabController/ViewController.swift:
```diff
@@ -0,0 +1,162 @@
+            $0.trailing.equalToSuperview().inset(42)
+        }
+        topTealRectangle.snp.makeConstraints {
+            $0.width.equalTo(topOrangeRectangle.snp.width)
```
Убери .snp.width

VAmpersand commented on Module_2 Groupe_1/Danil/TabController/ViewController.swift:
```diff
@@ -0,0 +1,162 @@
+            $0.width.equalTo(topOrangeRectangle.snp.width)
+            $0.height.equalTo(topTealRectangle.snp.width).dividedBy(4)
+            
+            $0.trailing.equalTo(topOrangeRectangle.snp.trailing)
```
Убери .snp.trailing

VAmpersand commented on Module_2 Groupe_1/Danil/TabController/ViewController.swift:
```diff
@@ -0,0 +1,162 @@
+        
+        //нижний квадрат
+        bottomBlueBox.snp.makeConstraints {
+            $0.width.height.equalTo(topBlueBox)
```
size

VAmpersand commented on Module_2 Groupe_1/Danil/TabController/ViewController.swift:
```diff
@@ -0,0 +1,162 @@
+            $0.width.height.equalToSuperview().multipliedBy(0.32)
+            
+            $0.top.leading.equalToSuperview().inset(16)
+        }
```
Вот тут стоит добавить отступ

VAmpersand commented on Module_2 Groupe_1/Danil/TabController/ViewController.swift:
```diff
@@ -0,0 +1,162 @@
+            
+            $0.centerY.equalToSuperview()
+            $0.trailing.equalToSuperview().inset(42)
+        }
```
Добавь отступ

VAmpersand commented on Module_2 Groupe_1/Danil/TabController/ViewController.swift:
```diff
@@ -0,0 +1,162 @@
+        bottomBlueBox.snp.makeConstraints {
+            $0.width.height.equalTo(topBlueBox)
+            $0.bottom.trailing.equalToSuperview().inset(16)
+        }
```
Добавь отступ

VAmpersand commented on Module_2 Groupe_1/Danil/TabController/ViewController.swift:
```diff
@@ -0,0 +1,162 @@
+        bottomOrangeRectangle.snp.makeConstraints {
+            $0.width.height.equalTo(topOrangeRectangle)
+            $0.leading.centerY.equalToSuperview()
+        }
```
Добавь отступ

VAmpersand commented on Module_2 Groupe_1/Danil/TabController/ViewController.swift:
```diff
@@ -0,0 +1,162 @@
+            $0.bottom.trailing.equalToSuperview().inset(16)
+        }
+        bottomOrangeRectangle.snp.makeConstraints {
+            $0.width.height.equalTo(topOrangeRectangle)
```
size

VAmpersand commented on Module_2 Groupe_1/Danil/TabController/ViewController.swift:
```diff
@@ -0,0 +1,162 @@
+            $0.leading.centerY.equalToSuperview()
+        }
+        bottomTealRectangle.snp.makeConstraints {
+            $0.width.height.equalTo(topTealRectangle)
```
size

VAmpersand commented on Module_2 Groupe_1/Danil/TabController/ViewController.swift:
```diff
@@ -0,0 +1,162 @@
+        bottomTealRectangle.snp.makeConstraints {
+            $0.width.height.equalTo(topTealRectangle)
+            $0.trailing.equalTo(bottomBlueBox)
+            $0.top.equalTo(bottomOrangeRectangle.snp.top)
```
Убери .snp.top

VAmpersand commented on Module_2 Groupe_1/Danil/TabController/ViewController.swift:
```diff
@@ -0,0 +1,162 @@
+    }
+}
+
+//можно закинуть в отдельный файл
```
Можно, создай папку Helpers в директории проекта и сделай 'UIView + ext' засунь туда это расширение. Мб для других ребят будет полезно)

VAmpersand commented on Module_2 Groupe_1/Danil/TabController/ViewController.swift:
```diff
@@ -0,0 +1,162 @@
+        return view
+    }()
+    
+    
```
Лишний отступ.

VAmpersand commented on Module_2 Groupe_1/Danil/TabController/ViewController.swift:
```diff
@@ -0,0 +1,162 @@
+    private let yellowRectangle = UIView(backgroundColor: .yellow)
+    private let greenRectangle = UIView(backgroundColor: .green)
+    
+    //top
```
Не знаю на сколько полезны такие комментерии, если ты и так неймингом указываешь top и bottom view

VAmpersand commented on Module_2 Groupe_1/Danil/TabController/ViewController.swift:
```diff
@@ -0,0 +1,162 @@
+    private func addConstrains() {
+        NSLayoutConstraint.activate(
+            [
+                // 16 можно вынести в константу
```
Комменеты как и код давай писать на англ, но они не нужны для таких очевидных блоков

danildemchenko committed:
fixes according to comments

1. got rid of "magic numbers"
2. add extension to the separate file
3. got rid of unnecessary comments and spaces

VAmpersand commented on Module_2 Groupe_1/Danil/TabController/ViewController.swift:
```diff
@@ -0,0 +1,147 @@
+import UIKit
+import SnapKit
+
+final class ViewController: UIViewController {
```
Заметил позже. Дай внятное название контроллеру.

danildemchenko committed:
change the name of controller

danildemchenko committed:
fixes

1. rename the controller
2. add the extension for constants
